File: server/src/main/java/hu/elte/alkfejl/entity/Task.java
package hu.elte.alkfejl.entity;

import java.util.List;
import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;


@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class Task extends BaseEntity {
    @Column(nullable = false)
    private String text;
    
    @Column(nullable = false)
    private String description;
    
    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private Progress progress;
   
    public enum Progress {
        BACKLOG, IN_PROGRESS, TEST, BLOCKED, DONE
    }

    //@JoinColumn
    @JsonIgnore
    @OneToOne(targetEntity = User.class)
    private User assignee;
    
    @JoinColumn
    @ManyToOne(targetEntity = Team.class)
    @JsonIgnore
    private Team team;

}

File: client-app/src/app/classes/task.ts
import { User } from './user';
import { Team } from './team';

export enum Progress {
    BACKLOG, IN_PROGRESS, TEST, BLOCKED, DONE
  }

export class Task {

    public id: number;
    private description: String;
    public progress: Progress;
    public prog: String
    private user: User = null;
    //private team: Team = null;
    
    public constructor(
      description?: String, prog?: String, user?: User, id?: number/*, team?: Team*/
    ) {
      this.description = description;
      this.prog = prog;
      //this.progress = progress;
      this.user = user;
      //this.team = team;
    }

    public getId(): number {
        return this.id;
    }

    public changeProgress(prog: Progress): void {
        this.progress = prog;
    }

    public toString(): string {
        return this.description + ' \n ' + this.progress;
    }

    public getProgress(): Progress {
        return this.progress;
    }
    
}

